test/src/main/java/org/apache/accumulo/test/ComprehensiveTableOperationsIT.java

```diff
@@ -0,0 +1,957 @@
+    // offline,online,isOnline not tested for system tables. Test basic functionality here
+    for (var sysTable : AccumuloTable.values()) {
+      switch (sysTable) {
+        case ROOT:
+        case METADATA:
+          assertTrue(client.tableOperations().isOnline(sysTable.tableName()));
+          assertThrows(AccumuloException.class,
+              () -> client.tableOperations().offline(sysTable.tableName(), true));
+          assertTrue(client.tableOperations().isOnline(sysTable.tableName()));
+          client.tableOperations().online(sysTable.tableName(), true);
+          break;
+        case SCAN_REF:
+        case FATE:
+          assertTrue(client.tableOperations().isOnline(sysTable.tableName()));
+          client.tableOperations().offline(sysTable.tableName(), true);
+          assertFalse(client.tableOperations().isOnline(sysTable.tableName()));
+          client.tableOperations().online(sysTable.tableName(), true);
+          assertTrue(client.tableOperations().isOnline(sysTable.tableName()));
+          break;
+        default:
+          throw new IllegalStateException("Unrecognized table: " + sysTable);
+      }
+    }
+  }
+
+  @Test
+  public void test_clearLocatorCache() throws Exception {
+    // clearLocatorCache is tested elsewhere for user tables but not for system tables. Simple
+    // enough to test all tables here
+    var clientContext = (ClientContext) client;
+    userTable = getUniqueNames(1)[0];
+
+    client.tableOperations().create(userTable);
+    ReadWriteIT.ingest(clientContext, 5, 5, 5, 0, userTable);
+    assertTrue(clientContext.getTabletLocationCache(TableId.of(ops.tableIdMap().get(userTable)))
+        .getTabletHostingRequestCount() > 0);
+    ops.clearLocatorCache(userTable);
+    assertEquals(0,
+        clientContext.getTabletLocationCache(TableId.of(ops.tableIdMap().get(userTable)))
+            .getTabletHostingRequestCount());
+
+    for (var sysTable : AccumuloTable.values()) {
+      assertEquals(0,
+          clientContext.getTabletLocationCache(sysTable.tableId()).getTabletHostingRequestCount());
+      ops.clearLocatorCache(sysTable.tableName());
+      assertEquals(0,
+          clientContext.getTabletLocationCache(sysTable.tableId()).getTabletHostingRequestCount());
+    }
+  }
+
+  @Test
+  public void
+      test_attachIterator_removeIterator_getIteratorSetting_listIterators_checkIteratorConflicts()
+          throws Exception {
+    // These may be tested elsewhere across several tests, but simpler to test all here
+    userTable = getUniqueNames(1)[0];
+    ops.create(userTable);
+    String iterName = "iter_name";
+    var iterSetting = new IteratorSetting(100, iterName, NoOpFilter.class);
+    var scope = EnumSet.of(IteratorUtil.IteratorScope.majc, IteratorUtil.IteratorScope.minc);
+    // system and user tables
+    Set<String> allTables = ops.list();
+
+    for (String table : allTables) {
+      ops.attachIterator(table, iterSetting, scope);
```

Comment:
Should custom iterators on the system tables continue to be allowed or is this undesired?

Reply:
That could be dangerous, but I think its fine to leave it because there are many other dangerous things that could potentially be done.   Overall the best option is to ensure the metadata table has locked down permissions for setting table properties and writing to it.